arq/worker.py

```diff
@@ -158,6 +161,7 @@ def __init__(
         job_timeout: SecondsTimedelta = 300,
         keep_result: SecondsTimedelta = 3600,
         poll_delay: SecondsTimedelta = 0.5,
+        queue_read_limit: Optional[int] = None,
```

Comment:
maybe this shouldn't be optional?

Is there any reason not to set it to `1000` or `max_jobs`?

Reply:
@rubik did you see this comment? What do you think?

Reply:
@samuelcolvin I agree, yes. One of the last commits sets it to `max_jobs` if it's None.

Reply:
okay, so the type shouldn't be `Optional`, on the class

should be:
```py
self.queue_read_limit: int = queue_read_limit or max_jobs
```

Reply:
@samuelcolvin I don't follow. In your line with `or` you allow `queue_read_limit` to be `None`. That means that we need to keep `Optional` in the type hint. If the user does not specify it, it's `None`. I changed the assignment to be a one-liner like yours.

Reply:
What you've done is correct.

My point is that if you do:

```py
foo: Optional[int] = 1

bar = foo

if bar is None:
    bar = 123
```
Mypy reads this as implicitly setting the type of `bar` to `Optional[int]`, that doesn't change in the `if` block.

If instead you do

```py
foo: Optional[int] = 1
bar = foo or 123
```
The type of `bar` is `int`, this is what we do above, except we're explicit and add a type hint.

Reply:
@samuelcolvin Now I understand, thanks for the explanation.